documentation/terraform_example.md

```diff
@@ -0,0 +1,448 @@
+            "ip_prefix_length": 16,
+            "ipc_mode": null,
+            "labels": null,
+            "links": null,
+            "log_driver": "json-file",
+            "log_opts": null,
+            "logs": false,
+            "max_retry_count": null,
+            "memory": null,
+            "memory_swap": null,
+            "mounts": [],
+            "must_run": true,
+            "name": "docker_target_0",
+            "network_alias": null,
+            "network_data": [
+              {
+                "gateway": "172.17.0.1",
+                "ip_address": "172.17.0.4",
+                "ip_prefix_length": 16,
+                "network_name": "bridge"
+              }
+            ],
+            "network_mode": null,
+            "networks": null,
+            "networks_advanced": [],
+            "pid_mode": null,
+            "ports": [
+              {
+                "external": 2200,
+                "internal": 22,
+                "ip": "0.0.0.0",
+                "protocol": "tcp"
+              }
+            ],
+            "privileged": null,
+            "publish_all_ports": null,
+            "restart": "no",
+            "rm": false,
+            "shm_size": null,
+            "start": true,
+            "sysctls": null,
+            "tmpfs": null,
+            "ulimit": [],
+            "upload": [],
+            "user": null,
+            "userns_mode": null,
+            "volumes": [],
+            "working_dir": null
+          },
+          "private": "eyJzY2hlbWFfdmVyc2lvbiI6IjEifQ==",
+          "depends_on": [
+            "docker_image.sshd"
+          ]
+        },
+```
+### Invnetory
+
+Show how to map values in example snipit to this (showing remote, could do statefile)
+```yaml
+version: 2
+groups:
+  - name: terraform-group
+    targets:
+      - _plugin: terraform
+        dir: /home/cas/working_dir/dynamic-inventory-demo
+        resource_type: docker_container.sshd.*
+        # statefile: test/terraform_local.tfstate
+        backend: remote
+        uri: network_data.0.gateway
+        name: name
+        config:
+          ssh:
+            port: ports.0.external
+    config:
+      transport: ssh
+      ssh:
+        user: root
+        password: root
+```
+try it out
+```
+cas@cas-ThinkPad-T460p:~/working_dir/bolt$ PATH=$PATH:~/working_dir/dynamic-inventory-demo/terraform bolt command run hostname -t terraform-group
+Started on docker_target_0...
+Started on docker_target_2...
+Started on docker_target_1...
+Finished on docker_target_0:
+  STDOUT:
+    f04670444f71
+Finished on docker_target_2:
+  STDOUT:
+    268e5d2572d7
+Finished on docker_target_1:
+  STDOUT:
+    3e32b6d85cfd
+Successful on 3 nodes: docker_target_0,docker_target_1,docker_target_2
+Ran on 3 nodes in 0.27 sec
+```
+
+### PKCS7 plugin
+Now pretend we want to protect that ssh password
+
+Configure pkcs7 in bolt.yaml
+```yaml
+plugins:
+  pkcs7:
+    private-key: ~/working_dir/dynamic-inventory-demo/keys/public.pkcs7.pem
```

Comment:
This is where setting up an explicit dir ahead of time for bolt.yaml and inventory.yaml would be helpful, could just say "now create `keys` dir....".